packages/db/postgres/tables.sql

```diff
@@ -299,17 +322,35 @@ CREATE TABLE IF NOT EXISTS metric
     updated_at  TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL
 );
 
-CREATE VIEW admin_search as
-select
-  u.id::text as user_id,
-  u.email as email,
-  ak.secret as token,
-  ak.id::text as token_id,
-  ak.deleted_at as deleted_at,
-  akh.inserted_at as reason_inserted_at,
-  akh.reason as reason,
-  akh.status as status
-from public.user u
-right join auth_key ak on ak.user_id = u.id
-full outer join (select * from auth_key_history where deleted_at is null) as akh on akh.auth_key_id = ak.id
-where ak.deleted_at is NULL or ak.deleted_at is not NULL and akh.status is not NULL;
+DO 
+$$
+BEGIN
+  IF NOT EXISTS (SELECT pg_get_viewdef('admin_search', true)) THEN
+    CREATE VIEW admin_search as
+    select
+      u.id::text as user_id,
+      u.email as email,
+      ak.secret as token,
+      ak.id::text as token_id,
+      ak.deleted_at as deleted_at,
+      akh.inserted_at as reason_inserted_at,
+      akh.reason as reason,
+      akh.status as status
+    from public.user u
+    right join auth_key ak on ak.user_id = u.id
+    full outer join (select * from auth_key_history where deleted_at is null) as akh on akh.auth_key_id = ak.id
+    where ak.deleted_at is NULL or ak.deleted_at is not NULL and akh.status is not NULL;
+  END IF;
+END
+$$;
+
+CREATE TABLE IF NOT EXISTS email_notification_history 
+(
+  id              BIGSERIAL PRIMARY KEY,
+  -- the id of the user being notified
+  user_id         BIGINT NOT NULL REFERENCES public.user (id),
+  email_type      email_type NOT NULL,
+  -- the email service messages unique id
+  message_id      TEXT NOT NULL,
+  sent_at         TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL
```

Comment:
I wonder if it be nice the store the type of services used to send the email. 
I'm sure, thinking if this is needed or not, can be part of  @adamalton work on the email client.